modules/abstract-lightning/src/wallet/lightning.ts

```diff
@@ -244,8 +256,15 @@ export class LightningWallet implements ILightningWallet {
       throw new Error(`Invalid list invoices response ${error}`);
     });
   }
-
-  async payInvoice(params: SubmitPaymentParams): Promise<PayInvoiceResponse> {
+  /**
+   * Private method to handle direct invoice payments
+   * @param {SubmitPaymentParams} params - Invoice payment parameters
+   * @returns {Promise<PayInvoiceResponse>} Payment result
+   * @private
+   */
+  private async payInvoice(
+    params: t.TypeOf<typeof SubmitPaymentParams> & { invoice: string }
```

Comment:
```suggestion
    params: t.TypeOf<typeof SubmitPaymentParams> & { invoice: string }
```

Why are we doing t.TypeOf? We aleady have SubmitPaymentParams as a type. Maybe what you should do instead is have a 2 codecs, one for lnurl and another for invoice and have SubmiyPaymentParams be an intersection of both and then payInvoice and payLnUrl can use the respective type